File: src/modes/default/src/main/java/work/lclpnet/ap2/mode_default/activity/ArcadePartyStartingActivity.java
package work.lclpnet.ap2.mode_default.activity;

import org.jetbrains.annotations.NotNull;
import org.jspecify.annotations.NonNull;
import org.slf4j.Logger;
import work.lclpnet.activity.ComponentActivity;
import work.lclpnet.activity.component.ComponentBundle;
import work.lclpnet.activity.component.builtin.BuiltinComponents;
import work.lclpnet.ap2.api.game.MiniGame;
import work.lclpnet.game.api.start.GameStartArgs;
import work.lclpnet.game.api.start.ItemReservationManager;
import work.lclpnet.game.impl.Voting;
import work.lclpnet.game.util.GameStartUtil;
import work.lclpnet.kibu.hook.HookRegistrar;
import work.lclpnet.kibu.scheduler.Ticks;

public class ArcadePartyStartingActivity extends ComponentActivity {

    private final GameStartArgs args;
    private final Voting<MiniGame> miniGameVoting;

    public ArcadePartyStartingActivity(@NonNull GameStartArgs args, @NonNull Logger logger, Voting<MiniGame> miniGameVoting) {
        super(args.options().getContext().getServer(), logger);

        this.args = args;
        this.miniGameVoting = miniGameVoting;
    }

    @Override
    protected void registerComponents(@NotNull ComponentBundle components) {
        components.add(BuiltinComponents.HOOKS);
    }

    @Override
    public void start() {
        super.start();

        ItemReservationManager.Reservation votingItemSlot = args.itemManager().reserve(4);

        if (votingItemSlot != null) {
            HookRegistrar hooks = component(BuiltinComponents.HOOKS).hooks();

            GameStartUtil.setupVoting(miniGameVoting, hooks, votingItemSlot.slot(), args, Ticks.seconds(15));
        }
    }
}

File: src/modes/default/src/main/kotlin/work/lclpnet/ap2/mode_default/activity/ArcadePartyStartingActivity.kt
package work.lclpnet.ap2.mode_default.activity

import org.slf4j.Logger
import work.lclpnet.activity.ComponentActivity
import work.lclpnet.activity.component.ComponentBundle
import work.lclpnet.activity.component.builtin.BuiltinComponents
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.game.api.start.GameStartArgs
import work.lclpnet.game.impl.Voting
import work.lclpnet.game.util.GameStartUtil
import work.lclpnet.kibu.scheduler.Ticks

class ArcadePartyStartingActivity(
    private val args: GameStartArgs,
    logger: Logger,
    private val miniGameVoting: Voting<MiniGame>
) : ComponentActivity(
    args.options().context.server,
    logger
) {

    override fun registerComponents(components: ComponentBundle) {
        components.add(BuiltinComponents.HOOKS)
    }

    override fun start() {
        super.start()

        val votingItemSlot = args.itemManager().reserve(4)

        if (votingItemSlot != null) {
            val hooks = component(BuiltinComponents.HOOKS).hooks()

            GameStartUtil.setupVoting(miniGameVoting, hooks, votingItemSlot.slot(), args, Ticks.seconds(15))
        }
    }
}